app/src/main/java/org/fossasia/openevent/adapters/LocationsListAdapter.java

```diff
@@ -105,18 +107,27 @@ public void moveItem(int fromPosition, int toPosition) {
         notifyItemMoved(fromPosition, toPosition);
     }
 
-    class Viewholder extends RecyclerView.ViewHolder {
+    public static interface AdapterOnClickHandler {
+        void onClick(int pos, Viewholder vh);
+    }
+
+    public static class Viewholder extends RecyclerView.ViewHolder implements View.OnClickListener {
         TextView name;
         TextView floor;
 
         public Viewholder(View itemView) {
             super(itemView);
             itemView.setClickable(true);
-
+            itemView.setOnClickListener(this);
             name = (TextView) itemView.findViewById(R.id.location_name);
             floor = (TextView) itemView.findViewById(R.id.location_floor);
 
         }
 
+        @Override
+        public void onClick(View view) {
+            clickHandler.onClick(getAdapterPosition(), this);
```

Comment:
Also I see that you may need access to the viewHolder  properties so you can also cache this listener as `final` in the ViewHolder class. So to re-iterate:
* Outer adapter caches the listener
* pass the listener in the viewHolder, either by setting it explicitly or attaching it like `viewHolder.setItemClickListener(cachedListener)`
* In the viewHolder cache this and then set the listener on the view. like `itemView.setListener(this); this.cachedListener = cachedListener;`
* The invoke the cachedListener during `onClick()` like: `this.cachedListener(getAdapterPosition(), view)`

The idea is something similar that you have already done except we don't create any new instances anywhere just pass along memory references to an already existing click listener.

This is important because a recycler view can contain items of different types which means the viewHolders will be different implementations. So if that happens you need to maintain these different implementations in separate java files instead of the Adapter and therefore it is good practice to have a structure early on. Since your viewHolder class is static you can use it in other adapters too! Facilitates reusability and tones down complexity and coupling between the components.

So if you are stuck about thinking how you can structure the code, think about this: How can I make sure this viewHolder can be used by a different adapter, without creating any new clickListeners anywhere else in the code.